src/snowcli/config.py

```diff
@@ -39,63 +84,40 @@ def _add_options(self):
             parse_str=tomlkit.parse,
         )
 
-    def get_section(self, *path) -> TOMLDocument:
+    def _find_section(self, *path) -> TOMLDocument:
         section = self
         idx = 0
         while idx < len(path):
             section = section[path[idx]]
             idx += 1
         return section
 
-    def section_exists(self, *path) -> bool:
-        try:
-            self.get_section(*path)
-            return True
-        except NonExistentKey:
-            return False
+    def _merge_section_with_env(self, section: Table, *path) -> Dict[str, str]:
+        merged_dict = {}
+        for k in section:
+            value = self._get_from_env(*path, key=k)
+            if value:
+                merged_dict[k] = value
+            else:
+                merged_dict[k] = section[k]
+        return merged_dict
```

Comment:
Maybe i'm missing something, but what happens if user won't put the required key in toml? In integration tests config, there's a dummy config, that holds required values.
So, if there's no "account" field in the config file, do we catch it during config parsing, or do we rely on python connector's messages?

Reply:
Paweł asked same question https://github.com/Snowflake-Labs/snowcli/pull/222#discussion_r1260937675